models/src/main/protobuf/CasperMessage.proto

```diff
@@ -79,12 +79,17 @@ message BlockMessageProto {
   int64              seqNum         =  6; // number of blocks created by the validator
   bytes              preStateHash   =  7; // hash of the tuplespace contents before new deploys
   bytes              postStateHash  =  8; // hash of the tuplespace contents after new deploys
-  repeated bytes     justifications =  9 [(scalapb.field).collection_type="collection.immutable.List"]; // map of all validators to latest blocks based on current view
-  repeated BondProto bonds          = 10 [(scalapb.field).collection_type="collection.immutable.List"]; // validators and their stakes
-  repeated bytes rejectedDeploys    = 11 [(scalapb.field).collection_type="collection.immutable.List"];
-  RholangStateProto  state          = 12 [(scalapb.field).no_box = true];
-  string             sigAlgorithm   = 13; // name of the algorithm used to sign
-  bytes              sig            = 14; // signature generated by signing blockHash
+  repeated bytes     justifications =  9 [(scalapb.field).collection_type="List"]; // map of all validators to latest blocks based on current view
+  repeated BondProto bonds          = 10 [(scalapb.field).collection_type="List"]; // validators and their stakes
+  // Rejected data in finalized fringe
+  repeated bytes rejectedDeploys    = 11 [(scalapb.field).collection_type="List"];
+  repeated bytes rejectedBlocks     = 12 [(scalapb.field).collection_type="List"];
+  repeated bytes rejectedSenders    = 13 [(scalapb.field).collection_type="List"];
```

Comment:
why do you want to put these 3 in a block? and what are `rejectedBlocks` and `rejectedSenders `?

Reply:
These fields store data for consensus algorithm and are related to the finalized fringe seen by the block. It's not necessary for the node to store this data in the block because it will have it in internal state when running consensus algorithm, but for LFS node which is starting from a fringe it will be used to create the next fringe after that.

1. **Rejected deploys** - deploys rejected in the fringe
2. **Rejected blocks** - blocks that are excluded from giving stake to finalized the fringe (invalid blocks)
3. **Rejected senders** - validators excluded from giving stake to finalized the fringe, but without having invalid e.g. lazy validator not producing new blocks